### DISCUSSION TITLE
How to retrieve the deleted forked repository

### DISCUSSION THREAD

**Discussion by ygg-amal** [community/community#142569] [Repositories]
### Select Topic Area

Question

### Body

Dear Team,

Due to a permissions change, the forked repository of one of our two members was deleted. We need to recover this forked repository as it contains important code our team has stored.

Could you please advise on how we can retrieve it?

**Comment by Aninooo** [comment]
- Check the trash/repositories section or Check for Local Clones. If any team members have cloned the repository locally, they can push their local version back to a new repository on the same platform.

**Comment by ygg-amal** [comment]
@Aninooo Each team member has stored some of their changes locally in their forks, and we need to recover these. Is there any possibility to retrieve the deleted fork or recover the changes from their local repositories?

**Reply by Aninooo** [reply]
Each team member can check their local clone of the deleted fork for any unpushed changes. They can use the following Git command to view their commit history
- git log 

This command will display a list of all commits made in the local repository, along with the time and date for each commit. This will help team members identify any changes that haven’t been pushed to the remote repository. they can copy the relevant commit history to restore their projects.

-If they have made commits locally but haven't pushed them, they can use git reflog to find previous states of their branches:
git reflog

**Comment by ygg-amal** [comment]
@Aninooo This solution helped one team member, but another team member made changes locally and pushed those changes to different branches in the deleted forked repository. Is there any way to retrieve those changes?

**Reply by Aninooo** [reply]
If the team member still has a local clone of the deleted fork, they can check all branches in their local repository by typing: git branch -a and After identifying the branches, they can switch to the desired branch using git checkout <branch-name>
and the use the git log This will allow them to view the commit history for that branch, following the same process as mentioned earlier.

**Comment by ygg-amal** [comment]
@Aninooo  I’m incredibly grateful for your help in resolving the issue! I can't thank you enough. Thank you very much!

**Reply by Aninooo** [reply]
Welcome sir
